one/src/daemon.rs

```diff
@@ -386,6 +435,43 @@ pub async fn run(opts: DaemonOpts) -> Result<()> {
         None
     };
 
+    // Start anchoring if remote anchor service URL is provided
+    let anchor_service_handle =
+        if let Some(remote_anchor_service_url) = opts.remote_anchor_service_url {
+            info!(
+                node_did = node_id.did_key(),
+                url = remote_anchor_service_url,
+                poll_interval = opts.anchor_poll_interval,
+                "starting remote cas anchor service"
+            );
+            let remote_cas = RemoteCas::new(
+                node_id,
+                keypair,
+                remote_anchor_service_url,
+                Duration::from_secs(opts.anchor_poll_interval),
+                opts.anchor_poll_retry_count,
+            );
+            let mut anchor_service = AnchorService::new(
+                Arc::new(remote_cas),
+                event_svc.clone(),
+                sqlite_pool.clone(),
+                node_id,
+                Duration::from_secs(opts.anchor_interval),
+                opts.anchor_batch_size,
+            );
```

Comment:
I really like how simple an straightforward this dependency injection is working. Create a service using its new method and pass it references of the deps.